### ISSUE TITLE
Error in `arrange()`:

### ISSUE THREAD

**Issue by wangsky137** [kharchenkolab/numbat#63]
I am trying to run numbat using my own data with matrix and allele built following instruction I got 
Error in `arrange()`:
! Problem with the implicit `transmute()` step.
✖ Problem while computing `..1 = CHROM`.
Caused by error in `mask$eval_all_mutate()`:
! object 'CHROM' not found
Run `rlang::last_error()` to see where the error occurred.
Warning message:
In mclapply(groups, mc.cores = ncores, function(g) { :
  all scheduled cores encountered errors in user code

**Comment by evanbiederstedt** [comment]
@wangsky137 

Thanks for the bug report

You'll need to provide a lot more information for us to fix this. 

What was the data you used? 
What were the commands you ran?
What does `sessionInfo()` give in your R session?

**Comment by wangsky137** [comment]
I am using the run_numbat function with a smartseq input 
 out = run_numbat(
    count_mat_ATC2, # gene x cell integer UMI count matrix 
    ref_hca, # reference expression profile, a gene x cell type normalized expression level matrix
    df_allele_ATC2, # allele dataframe generated by pileup_and_phase script
    genome = "hg38",
    t = 1e-5,
    ncores = 4,
    plot = TRUE,
    gamma=5,
    out_dir = './'
  )

I have also attached the log.txt
INFO [2022-11-22 12:48:59] Running under parameters:
t = 1e-05
alpha = 1e-04
gamma = 5
min_cells = 50
init_k = 3
max_cost = 91.5
max_iter = 2
max_nni = 100
min_depth = 0
use_loh = auto
multi_allelic = TRUE
min_LLR = 5
min_overlap = 0.45
max_entropy = 0.5
skip_nj = FALSE
diploid_chroms = 
ncores = 4
ncores_nni = 4
common_diploid = TRUE
tau = 0.3
check_convergence = FALSE
plot = TRUE
genome = hg38
Input metrics:
305 cells
INFO [2022-11-22 12:49:01] Mem used: 1.87Gb
INFO [2022-11-22 12:49:02] Approximating initial clusters using smoothed expression ..
INFO [2022-11-22 12:49:03] Mem used: 1.87Gb
INFO [2022-11-22 12:49:05] running hclust...
INFO [2022-11-22 12:49:23] Iteration 1
INFO [2022-11-22 12:49:25] Mem used: 2.23Gb
ERROR [2022-11-22 12:49:31] job 1,2,3,4,5 failed
ERROR [2022-11-22 12:49:31] Error in summarise(., AD = sum(AD), DP = sum(DP), AR = AD/DP, .groups = "drop") : 
  [1m[22mProblem while computing `DP = sum(DP)`.
[36mℹ[39m The error occurred in group 1: snp_id = "10_100000235_C_T", CHROM = 10, POS = 100000235, cM = "126.46143916246", REF = "C", ALT = "T", GT = "1|0", gene = "DNMBP".
[1mCaused by error in `sum()`:[22m
[33m![39m invalid 'type' (character) of argument

and here's sessionInfo output 

R version 4.0.2 (2020-06-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Red Hat Enterprise Linux

Matrix products: default
BLAS/LAPACK: /usr/local/lib64/libmkl_gf_lp64.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8      
 [8] LC_NAME=C                  LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] vcfR_1.13.0               data.table_1.14.2         glue_1.6.2                numbat_1.0.2              Matrix_1.5-1              forcats_0.5.2             stringr_1.4.1             purrr_0.3.5              
 [9] readr_2.1.3               tidyr_1.2.1               tibble_3.1.8              tidyverse_1.3.2           ggplot2_3.3.6             stxBrain.SeuratData_0.1.1 SeuratData_0.2.1          patchwork_1.1.2          
[17] dplyr_1.0.10              sp_1.5-0                  SeuratObject_4.1.2        Seurat_4.2.0              copykat_1.0.8            

loaded via a namespace (and not attached):
  [1] utf8_1.2.2                  reticulate_1.26             tidyselect_1.2.0            htmlwidgets_1.5.4           grid_4.0.2                  Rtsne_0.16                  munsell_0.5.0              
  [8] codetools_0.2-18            ica_1.0-3                   future_1.28.0               miniUI_0.1.1.1              withr_2.5.0                 argparse_2.1.6              spatstat.random_2.2-0      
 [15] colorspace_2.0-3            progressr_0.11.0            Biobase_2.48.0              logger_0.2.2                rstudioapi_0.14             stats4_4.0.2                SingleCellExperiment_1.10.1
 [22] ROCR_1.0-11                 tensor_1.5                  listenv_0.8.0               GenomeInfoDbData_1.2.3      polyclip_1.10-0             farver_2.1.1                TH.data_1.1-1              
 [29] coda_0.19-4                 parallelly_1.32.1           vctrs_0.4.2                 treeio_1.12.0               generics_0.1.3              lambda.r_1.2.4              xfun_0.33                  
 [36] fastcluster_1.2.3           GenomeInfoDb_1.24.2         R6_2.5.1                    doParallel_1.0.17           graphlayouts_0.8.2          locfit_1.5-9.4              reshape_0.8.9              
 [43] DelayedArray_0.14.1         bitops_1.0-7                spatstat.utils_2.3-1        gridGraphics_0.5-1          assertthat_0.2.1            promises_1.2.0.1            scales_1.2.1               
 [50] multcomp_1.4-20             pinfsc50_1.2.0              ggraph_2.0.6                googlesheets4_1.0.1         rgeos_0.5-9                 gtable_0.3.1                globals_0.16.1             
 [57] goftest_1.2-3               sandwich_3.0-2              tidygraph_1.2.2             rlang_1.0.6                 splines_4.0.2               lazyeval_0.2.2              gargle_1.2.1               
 [64] rjags_4-13                  spatstat.geom_2.4-0         broom_1.0.1                 BiocManager_1.30.18         reshape2_1.4.4              abind_1.4-5                 modelr_0.1.9               
 [71] backports_1.4.1             httpuv_1.6.6                tools_4.0.2                 ggplotify_0.1.0             gplots_3.1.3                ellipsis_0.3.2              spatstat.core_2.4-4        
 [78] RColorBrewer_1.1-3          BiocGenerics_0.34.0         ggridges_0.5.4              Rcpp_1.0.9                  plyr_1.8.7                  zlibbioc_1.34.0             RCurl_1.98-1.9             
 [85] rpart_4.1.16                deldir_1.0-6                pbapply_1.5-0               viridis_0.6.2               cowplot_1.1.1               S4Vectors_0.26.1            zoo_1.8-11                 
 [92] SummarizedExperiment_1.18.2 haven_2.5.1                 ggrepel_0.9.2               cluster_2.1.4               fs_1.5.2                    tinytex_0.42                infercnv_1.4.0             
 [99] magrittr_2.0.3              futile.options_1.0.1        scattermore_0.8             lmtest_0.9-40               reprex_2.0.2                RANN_2.6.1                  mvtnorm_1.1-3              
[106] googledrive_2.0.0           fitdistrplus_1.1-8          matrixStats_0.62.0          hms_1.1.2                   mime_0.12                   xtable_1.8-4                RhpcBLASctl_0.21-247.1     
[113] readxl_1.4.1                IRanges_2.22.2              gridExtra_2.3               compiler_4.0.2              KernSmooth_2.23-20          crayon_1.5.2                htmltools_0.5.3            
[120] ggfun_0.0.7                 mgcv_1.8-40                 later_1.3.0                 tzdb_0.3.0                  libcoin_1.0-9               aplot_0.1.8                 RcppParallel_5.1.5         
[127] lubridate_1.8.0             DBI_1.1.3                   formatR_1.12                tweenr_2.0.2                dbplyr_2.2.1                MASS_7.3-58.1               rappdirs_0.3.3             
[134] permute_0.9-7               cli_3.4.1                   parallel_4.0.2              igraph_1.3.5                GenomicRanges_1.40.0        pkgconfig_2.0.3             rvcheck_0.1.8              
[141] coin_1.4-2                  plotly_4.10.0               spatstat.sparse_2.1-1       xml2_1.3.3                  foreach_1.5.2               ggtree_2.2.4                XVector_0.28.0             
[148] rvest_1.0.3                 yulab.utils_0.0.5           digest_0.6.29               sctransform_0.3.5           RcppAnnoy_0.0.19            vegan_2.6-2                 spatstat.data_2.2-0        
[155] cellranger_1.1.0            leiden_0.4.3                tidytree_0.4.1              edgeR_3.30.3                uwot_0.1.14                 gtools_3.9.3                modeltools_0.2-23          
[162] shiny_1.7.2                 lifecycle_1.0.2             nlme_3.1-159                jsonlite_1.8.2              futile.logger_1.4.3         limma_3.44.3                viridisLite_0.4.1          
[169] fansi_1.0.3                 pillar_1.8.1                lattice_0.20-45             fastmap_1.1.0               httr_1.4.4                  survival_3.4-0              png_0.1-7                  
[176] iterators_1.0.14            ggforce_0.4.0               stringi_1.7.8               caTools_1.18.2              irlba_2.3.5.1               future.apply_1.9.1          ape_5.6-2

**Comment by teng-gao** [comment]
Hi, seems like there's something wrong with the format of the allele dataframe. Please double check the data types of each column with the example (df_allele_ATC2), especially for `DP` column, which error message says is character.

**Comment by wangsky137** [comment]
I have tried to cocerce the input dp as numeric but had the error NA were introduced. But still I got this error. should I go ahead and remove column with no input? 


Error in `arrange()`:
! Problem with the implicit `transmute()` step.
✖ Problem while computing `..1 = CHROM`.
Caused by error in `mask$eval_all_mutate()`:
! object 'CHROM' not found
Run `rlang::last_error()` to see where the error occurred.
Warning message:
In mclapply(groups, mc.cores = ncores, function(g) { :
  all scheduled cores encountered errors in user code


INFO [2022-11-28 10:38:18] Mem used: 1.25Gb
INFO [2022-11-28 10:38:18] Approximating initial clusters using smoothed expression ..
INFO [2022-11-28 10:38:19] Mem used: 1.25Gb
INFO [2022-11-28 10:38:22] running hclust...
INFO [2022-11-28 10:38:49] Iteration 1
INFO [2022-11-28 10:38:51] Mem used: 1.62Gb
ERROR [2022-11-28 10:38:58] job 1,2,3,4,5 failed
ERROR [2022-11-28 10:38:58] Error in mutate(., inter_snp_cm = c(NA, cM[2:length(cM)] - cM[1:(length(cM) -  : 
  [1m[22mProblem while computing `inter_snp_cm = c(NA, cM[2:length(cM)] - cM[1:(length(cM) - 1)])`.
[36mℹ[39m The error occurred in group 1: CHROM = 1.
[1mCaused by error in `cM[2:length(cM)] - cM[1:(length(cM) - 1)]`:[22m
[33m![39m non-numeric argument to binary operator




example of df_allele file 

         cell       snp_id CHROM    POS                cM REF ALT AD DP  GT      gene
1  SRR3500265 1_825532_C_T     1 825532  0.48995214890403   C   T  0  1 1|0          
2  SRR3500305 1_825532_C_T     1 825532  0.48995214890403   C   T  0  5 1|0          
3  SRR3500350 1_825532_C_T     1 825532  0.48995214890403   C   T  0 22 1|0          
4  SRR3500353 1_825532_C_T     1 825532  0.48995214890403   C   T  4  4 1|0          
5  SRR3500276 1_840279_A_G     1 840279 0.499038892588782   A   G  0  3 1|0 LINC01128
6  SRR3500292 1_840279_A_G     1 840279 0.499038892588782   A   G  0  2 1|0 LINC01128
7  SRR3500303 1_840279_A_G     1 840279 0.499038892588782   A   G  1  1 1|0 LINC01128
8  SRR3500328 1_840279_A_G     1 840279 0.499038892588782   A   G  3  3 1|0 LINC01128
9  SRR5168805 1_840279_A_G     1 840279 0.499038892588782   A   G 12 12 1|0 LINC01128
10 SRR3500276 1_841742_A_T     1 841742 0.499703768560114   A   T  0  2 1|0 LINC01128
11 SRR3500280 1_841742_A_T     1 841742 0.499703768560114   A   T  0  5 1|0 LINC01128
12 SRR3500283 1_841742_A_T     1 841742 0.499703768560114   A   T  0  4 1|0 LINC01128
13 SRR3500311 1_841742_A_T     1 841742 0.499703768560114   A   T  5  5 1|0 LINC01128
14 SRR3500310 1_841742_A_T     1 841742 0.499703768560114   A   T  6  6 1|0 LINC01128
15 SRR3500316 1_841742_A_T     1 841742 0.499703768560114   A   T  0 11 1|0 LINC01128
16 SRR3500323 1_841742_A_T     1 841742 0.499703768560114   A   T  0  6 1|0 LINC01128
17 SRR3500328 1_841742_A_T     1 841742 0.499703768560114   A   T  6  6 1|0 LINC01128
18 SRR3500333 1_841742_A_T     1 841742 0.499703768560114   A   T  0  8 1|0 LINC01128
19 SRR5168611 1_841742_A_T     1 841742 0.499703768560114   A   T  2  2 1|0 LINC01128
20 SRR5168679 1_841742_A_T     1 841742 0.499703768560114   A   T 20 20 1|0 LINC01128

**Comment by wangsky137** [comment]
<img width="910" alt="Screen Shot 2022-11-28 at 10 53 46 AM" src="https://user-images.githubusercontent.com/89659141/204335858-7fb4e055-8360-4c5d-aa37-73cfd7cb7111.png">

I think these line are causing problem?

**Comment by teng-gao** [comment]
Interesting .. that indeed looks like a bug in `pileup_and_phase`. Could you try again with the main github branch (v1.0.5) or the updated docker image and see if error persists?

**Comment by wangsky137** [comment]
The server I used is not compatible with docker or singularity so I didn't use the docker but just installed the software and run the pileup and phase.  could that be what causing the problem?

**Comment by teng-gao** [comment]
I don't think so, but you can get the most recent `pileup_and_phase.R` script using devtools:
```
devtools::install_github("https://github.com/kharchenkolab/numbat")
```

**Comment by teng-gao** [comment]
Fixed in v1.0.5
